Runtime/MeshOperations/AppendElements.cs

```diff
@@ -250,9 +260,18 @@ public static ActionResult CreateShapeFromPolygon(this ProBuilderMesh mesh, ILis
             {
                 int[] indexes = triangles.ToArray();
 
+                // check that all points are represented in the triangulation
+                HashSet<int> triangleSet = new HashSet<int>(triangles);
+                if(triangleSet.Count != vertices.Length)
+                {
+                    ClearAndRefreshMesh(mesh);
```

Comment:
<s>I don't think this should clear the mesh unless the operation succeeds. It should be up to the calling function to decide what to do in the case of failure.</s>

No, on second thought I think you're right. This is fine.